types/react-router-dom/index.d.ts

```diff
@@ -71,13 +71,17 @@ export interface Link<S = H.LocationState>
         React.PropsWithoutRef<LinkProps<S>> & React.RefAttributes<HTMLAnchorElement>
     > {}
 
-export interface NavLinkProps<S = H.LocationState> extends LinkProps<S> {
+export interface NavLinkProps<S = H.LocationState> extends Omit<LinkProps<S>, "className" | "style"> {
     activeClassName?: string | undefined;
     activeStyle?: React.CSSProperties | undefined;
     exact?: boolean | undefined;
     strict?: boolean | undefined;
     isActive?<Params extends { [K in keyof Params]?: string }>(match: match<Params> | null, location: H.Location<S>): boolean;
     location?: H.Location<S> | undefined;
+    className?: string | ((isActive: boolean) => string);
```

Comment:
According to the docs it should be `(props: { isActive: boolean })`, but in the reality it is `(isActive: boolean)`
I'm not sure if this PR should be merged before `react-router` devs fix this issue

Reply:
> According to the docs

Could you link the docs? I only found mention of this feature in the release notes.

We should sync with the react-router folks on what the intended interface is here. It's a bit odd that we document different types than are actually documented.

Reply:
Seems to be it's the only mention of this change. Anyways, I've created an issue https://github.com/remix-run/react-router/issues/8028. Lets wait until they decide how it should be fixed and then I'll update this PR if necessary

Reply:
@eps1lon https://github.com/remix-run/react-router/issues/8028#issuecomment-915212576 Seems to be this PR could be merged :)